internal/sql/pg/system.go

```diff
@@ -110,10 +166,29 @@ var settingValidations = map[string]settingValidFn{
 	"max_wal_senders":           wantMinIntFn(10),
 	"max_replication_slots":     wantMinIntFn(10),
 	"max_prepared_transactions": wantMinIntFn(2),
-
-	"synchronous_commit": wantOnFn(true),
-	"fsync":              wantOnFn(true),
-	"max_connections":    wantMinIntFn(50),
+	"wal_sender_timeout":        orValidFn(wantIntFn(0), wantMinIntFn(3_600_000)), // ms units, 0 for no limit or 1 hr min
+
+	// We shouldn't have idle abandoned transactions, but we need to investigate
+	// the effect of allowing this kind of clean up.
+	"idle_in_transaction_timeout": wantIntFn(0), // disable disable idle transaction timeout for now
+
+	// Behavior related settings that must be set properly for determinism
+	// https://www.postgresql.org/docs/16/runtime-config-compatible.html
+	// The following are the documented defaults, which we enforce.
+	"array_nulls":                 wantOnFn(true),                // recognize NULL in array parser, false is for pre-8.2 compat
+	"standard_conforming_strings": wantOnFn(true),                // backslashes (\) in string literals are treated literally -- only escape syntax (E'...') is still usable if needed
+	"transform_null_equals":       wantOnFn(false),               // do not treat "expr=NULL" as "expr IS NULL"
+	"backslash_quote":             wantStringFn("safe_encoding"), // reject escaped single quotes like \', require standard ''
```

Comment:
Just to clarify, these are the default postgresql behaviors that we are designing around.  These checks are here to ensure postgres is running with the expected settings.